components/event-publisher/org.wso2.carbon.event.publisher.core/src/main/java/org/wso2/carbon/event/publisher/core/internal/EventPublisher.java

```diff
@@ -280,4 +309,21 @@ public void process(Event event) {
     public org.wso2.siddhi.query.api.definition.StreamDefinition getStreamDefinition() {
         return streamDefinition;
     }
+
+    public class EventWrapper{
+
+        private Event event;
+        private long timestampInMilies;
+
+        public EventWrapper(Event event, long timestamp){
+            this.event = event;
+            this.timestampInMilies = timestamp;
+        }
+        public Event getEvent(){
+            return event;
+        }
+        public long getTimestampInMilies(){
+            return timestampInMilies;
+        }
+    }
```

Comment:
Thanks @nirmal070125, code is formatted in the pull request - https://github.com/wso2/carbon-analytics-common/pull/121